### PR TITLE
Fix/conda remove permissions

### LINKED ISSUE #1751: conda clean -pt as non-root user with root anaconda install
I have installed root miniconda at /opt/anaconda.  When running 
```
conda clean -pt
```
as a lesser user than root, I am seeing errors indicating conda is not checking permissions before attempting to delete package dirs:

```
conda clean -pt
Cache location: /opt/anaconda/pkgs
Will remove the following tarballs:

/opt/anaconda/pkgs
------------------
conda-3.18.3-py27_0.tar.bz2                  175 KB
conda-env-2.4.4-py27_0.tar.bz2                24 KB
itsdangerous-0.24-py27_0.tar.bz2              16 KB
markupsafe-0.23-py27_0.tar.bz2                30 KB
flask-0.10.1-py27_1.tar.bz2                  129 KB
jinja2-2.8-py27_0.tar.bz2                    263 KB
anaconda-build-0.12.0-py27_0.tar.bz2          69 KB
flask-wtf-0.8.4-py27_1.tar.bz2                12 KB
flask-ldap-login-0.3.0-py27_1.tar.bz2         13 KB

---------------------------------------------------
Total:                                       730 KB

removing conda-3.18.3-py27_0.tar.bz2
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/opt/anaconda/envs/anaconda.org/bin/conda", line 5, in <module>
    sys.exit(main())
  File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 195, in main
    args_func(args, p)
  File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 202, in args_func
    args.func(args, p)
  File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/main_clean.py", line 331, in execute
    rm_tarballs(args, pkgs_dirs, totalsize, verbose=not args.json)
  File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/main_clean.py", line 156, in rm_tarballs
    os.unlink(os.path.join(pkgs_dir, fn))
OSError: [Errno 13] Permission denied: '/opt/anaconda/pkgs/conda-3.18.3-py27_0.tar.bz2'

```

### PR DESCRIPTION
**Pull request by rafaelhbarros**
@jvestuto this includes your changes and the test fixes.

Closes #1751 

Addresses the permission issue when executing conda clean.
@groutr please comment.

### CODE REVIEW

File: conda/install.py [hunk 1]
```diff
@@ -138,6 +138,13 @@ def _remove_readonly(func, path, excinfo):
     os.chmod(path, stat.S_IWRITE)
     func(path)
 
+def warn_failed_remove(function, path, exc_info):
+    if exc_info[1].errno == errno.EACCES:
+        log.warn( "WARNING: cannot remove, permission denied: %s" % path )
```
Comment (groutr) [inline conda/install.py]: We don't need `WARNING` in log.warn calls.

**Comment by groutr** [comment]
Thanks Jason and Rafael!

**Comment by stephenakearns** [comment]
@groutr - @rafaelhbarros is out sick today so I was curious to find out what are the steps for us to access a version of conda with this fix in it?

**Comment by rafaelhbarros** [comment]
The fixes are in the master branch, it could be accessed on the next
release or in dev mode (read the contribution part of readme.md)

@groutr <https://github.com/groutr> - @rafaelhbarros
<https://github.com/rafaelhbarros> is out sick today so I was curious to
find out what are the steps for us to access a version of conda with this
fix in it?

—
Reply to this email directly or view it on GitHub
<https://github.com/conda/conda/pull/1807#issuecomment-156442183>.

**Pull request merged**